src/AddIns/Uno.UI.MediaPlayer.Skia.Gtk/GTKMediaPlayer.cs

```diff
@@ -184,20 +190,18 @@ private void UpdateVideoStretch()
 						}
 						var videoRatio = (double)videoHeight / (double)videoWidth;
 
-						var newHeight = (videoRatio > playerRatio) ? this.ActualHeight : this.ActualWidth * videoRatio;
-						var newWidth = (videoRatio > playerRatio) ? this.ActualHeight / videoRatio : this.ActualWidth;
-
+						var newHeight = (int)Math.Floor((videoRatio > playerRatio) ? playerHeight : playerWidth * Math.Round(videoRatio, 4));
+						var newWidth = (int)Math.Ceiling((videoRatio > playerRatio) ? (playerHeight / videoRatio) : playerWidth);
```

Comment:
Using Floor and Ceiling seems random and likely to cause transparent cells due to the video being the wrong size. Values should stay as double for all calculations until the values are used to set the size of the video

Reply:
Fixed.